test/gtest/ucs/test_conn_match.cc

```diff
@@ -141,44 +128,63 @@ UCS_TEST_F(test_conn_match, random_insert_retrieve) {
     const size_t        max_addresses      = 128;
     const ucs_conn_sn_t max_conns          = 128;
     const size_t        max_address_length = 2048 / ucs::test_time_multiplier();
-    const size_t        min_address_length =
-        ucs::to_string(std::numeric_limits<uint64_t>::max()).size();
     const size_t        max_iters          = 4;
 
     for (size_t it = 0; it < max_iters; it++) {
-        std::vector<std::vector<conn_elem_t> > elems(max_addresses);
-        size_t address_length = (ucs::rand() %
-                                 (max_address_length + 1 - min_address_length)) +
-                                min_address_length;
-
-        UCS_TEST_MESSAGE << "address length: " << address_length;
+        std::map<std::string, conn_peer_t*> peers;
+        size_t address_length = (ucs::rand() % max_address_length) + 1;
 
         init_new_address_length(address_length);
 
+        size_t unique_addresses = 0;
         for (size_t i = 0; i < max_addresses; i++) {
             ucs_conn_sn_t num_conns = (ucs::rand() % max_conns) + 1;
-            elems[i].resize(num_conns);
 
-            void *dest_address = generate_random_address(address_length);
+            std::string dest_address;
+            ucs::generate_random_str(dest_address, address_length);
```

Comment:
seems over complicated 
can't we just generate list of random addresses and then do random shuffle on it for the peers?

Reply:
then we should uuid to make sure that address is unique

Reply:
@yosefe simplified